### PR TITLE
move conversions to c

### PR DESCRIPTION
**Pull request by LandryNorris**
Perform conversions on data in the C code. This makes the raw node-rhpslib module more compatible with the control hub protocol.

### CODE REVIEW

**Review by jfabellera — APPROVED** [review]

**Review by NoahAndrews — APPROVED** [review]

File: packages/core/src/ExpansionHub.ts [hunk 1]
```diff
@@ -322,26 +324,62 @@ export interface ExpansionHub extends RevHub {
...
      * @param motorMode
-     * @param pid
+     * @param algorithm Legacy PID algorithm
+     * @param pid Legacy PID coefficients
      */
-    setMotorPIDCoefficients(
+    setMotorClosedLoopControlCoefficients(
         motorChannel: number,
-        motorMode: MotorMode,
+        motorMode: number,
```
Comment (NoahAndrews) [inline packages/core/src/ExpansionHub.ts:336]: This looks to be a regression

File: packages/core/src/ExpansionHub.ts [hunk 2]
```diff
@@ -322,26 +324,62 @@ export interface ExpansionHub extends RevHub {
...
+     * @param motorChannel
+     * @param motorMode
+     * @param algorithm either Legacy PID or PIDF algorithm.
+     * @param pidf either PidCoefficients or PidfCoefficients depending on algorithm.
+     */
+    setMotorClosedLoopControlCoefficients(
+        motorChannel: number,
+        motorMode: number,
+        algorithm: ClosedLoopControlAlgorithm,
+        pidf: any,
```
Comment (NoahAndrews) [inline packages/core/src/ExpansionHub.ts]: Is there a reason this is `any` instead of a union type?

File: packages/core/src/ExpansionHub.ts [hunk 3]
```diff
@@ -322,26 +324,62 @@ export interface ExpansionHub extends RevHub {
...
+     * Get motor closed loop control coefficients.
+     * Will return either PidCoefficients or PidfCoefficients depending on which
+     * algorithm is active.
+     * @param motorChannel
+     * @param motorMode
+     */
+    getMotorClosedLoopControlCoefficients(
+        motorChannel: number,
+        motorMode: number,
+    ): Promise<PidfCoefficients | PidCoefficients>;
```
Comment (NoahAndrews) [inline packages/core/src/ExpansionHub.ts:382]: Should explictily return an algorithm alongside the parameters.

**Review by NoahAndrews — DISMISSED** [review]

File: packages/core/src/ExpansionHub.ts (hunk 3 above)
Comment (LandryNorris) [reply packages/core/src/ExpansionHub.ts]: the coefficients classes have an algorithm property now.

**Pull request merged**